### PR TITLE
fix: stabilize rte debounce

### PR DESCRIPTION
**Pull request by SamCreasey**
The debounce function was being re-rendered every time which meant that the timeout was never cleared if the component rerendered, causing a memory leak and triggering the callback unintentionally

### CODE REVIEW

File: src/hooks/useDebounce.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import { useCallback, useRef } from 'react';
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export const useDebounce = <T extends (...args: any[]) => void>(func: T, timeout = 10) => {
+    const timer = useRef<number | undefined>();
+
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const debouncedFunc = useCallback((...args: any[]) => {
+        clearTimeout(timer.current);
+        timer.current = window.setTimeout(() => func.apply(this, args), timeout);
```
Comment (fulopdaniel) [inline src/hooks/useDebounce.ts:12]: `window` is omittable I think

File: src/utilities/index.ts [hunk 2]
```diff
@@ -10,7 +10,6 @@ export * from './focusStyle';
 export * from './generateRandomId';
 export * from './merge';
 export * from './number';
-export * from './useDebounce';
```
Comment (fulopdaniel) [inline src/utilities/index.ts:13]: can utilities/debounce be removed as well? Also can we change all implementations that use utilities/debounce use the new usedebounce hook?

File: src/hooks/useDebounce.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import { useCallback, useRef } from 'react';
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export const useDebounce = <T extends (...args: any[]) => void>(func: T, timeout = 10) => {
+    const timer = useRef<number | undefined>();
+
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const debouncedFunc = useCallback((...args: any[]) => {
```
[the remark this answers is not in the document]
Comment (SamCreasey) [reply src/hooks/useDebounce.ts]: fixed

File: src/hooks/useDebounce.ts (hunk 1 above)
Comment (SamCreasey) [reply src/hooks/useDebounce.ts]: fixed

File: src/utilities/index.ts (hunk 2 above)
Comment (SamCreasey) [reply src/utilities/index.ts:13]: not yet

**Review by fulopdaniel — APPROVED** [review]

**Pull request merged**